2_process/src/sites_along_waterbody.R

```diff
@@ -19,10 +18,39 @@ sites_along_waterbody <- function(sw_sites_sf, waterbody_sf, lake_waterbody = FA
       summarize(geometry = sf::st_union(geometry)) %>%
       sf::st_buffer(dist = units::set_units(250, m))
```

Comment:
Have you had any instances where this results in more than 1 stream segment being associated with a site?

Reply:
In `sites_along_waterbody` sites with a stream order 3+ are determined by a 250m buffer from the nearest stream. This is problematic because sites may be associated with many stream segments in the process, some that are stream order 3 or higher, but also possibly lower. In creating the `p2_sw_streamorder3_sites` target, 1317 sites were within the 250m buffer of more than one stream segment, with some intersecting as many as 8. Some of this error is addressed by using a categorical stream order variable, but it seems likely some sites are getting misclassified.

Reply:
```
tar_load(p2_site_in_watersheds_sf)
tar_load(p2_lake_tributaries)

waterbody_buffered <- p2_lake_tributaries %>%
  group_by(comid, streamorde) %>%
  summarize(geometry = sf::st_union(geometry)) %>%
  sf::st_buffer(dist = units::set_units(250, m))

st_join(p2_site_in_watersheds_sf, waterbody_buffered, left = FALSE) |>
  group_by(site_no)|>
  summarize(comids = length(unique(comid))) |>
  filter(comids > 1) |>
  arrange(desc(comids))
```

<img width="712" alt="image" src="https://user-images.githubusercontent.com/17803537/210025626-b86514e0-14bc-400f-83b5-53f907e02870.png">

Reply:
Right, so they could be counted as SO3+ if they fall on smaller streams nearby SO3+